[히히] 로또 2단계 미션 제출합니다!

yelimkim98 opened the pull request:
로또 2단계 미션 제출합니다! 감사합니다!

yelimkim98 committed:
docs: 수동 로또 기능목록 작성

yelimkim98 committed:
feat: ManualLottoController 클래스 추가

- 기존의 LottoController 클래스는
AutoLottoController 로 클래스이름이 변경되었음.

yelimkim98 committed:
feat: 로또를 일부는 수동, 일부는 자동으로 생성하는 기능 구현

yelimkim98 committed:
feat: 수동 생성할 로또 갯수 입력받기 기능 구현

yelimkim98 committed:
refactor: 수동로또 생성 메서드에서 가변인자 사용을 리스트 사용으로 바꿈

yelimkim98 committed:
refactor: 수동구매할 로또 갯수가 전체 로또갯수보다 클 때 에러처리 구현

- LottosFactory 의 수동로또들을 생성하는 메서드에서
에러를 처리하긴 하지만,
그러면 수동 로또번호를 모두 입력받은 다음에 오류를 내게되기 때문에
컨트롤러에서 별도의 에러처리를 하도록 구현함.

yelimkim98 committed:
feat: 수동입력할 로또번호들을 입력받고 구매한 모든 로또 출력 구현

yelimkim98 committed:
refactor: 구매금액이 1000원 미만일 때 에러처리 시점을 구매금액 입력 직후로 앞당김

yelimkim98 committed:
feat: lotto 도메인을 활용하여 우승로또 입력받기부터 끝까지 구현

yelimkim98 committed:
refactor: 메서드 하나의 길이를 줄이기 위해 private 메서드로 분리함

yelimkim98 committed:
refact: 메서드 하나의 길이를 줄이기 위해 private 메서드로 분리함

yelimkim98 committed:
Merge branch 'yelimkim98' of https://github.com/yelimkim98/java-lotto into yelimkim98

kingbbode commented on src/main/java/lotto/domain/LottosFactory.java:
```diff
@@ -12,27 +12,74 @@
  * 날짜 : 2020/02/21
  */
 public class LottosFactory {
```
`LottosFactory` 에서 제공해야할 데이터가 많네요 :)
전역 클래스로서 데이터를 소유하는 것이 아닌, 기능을 나열했기 때문인 것 같네요 :)
전역 클래스가 아닌, 객체로서 필요한 정보를 소유하고, 흐름을 제어할 수 있다면 불필요하게 많은 정보를 제공하지 않을 수 있을 것 같아요.

이 클래스에서 비지니스 로직만 처리하기 때문에 `ManualLottoController` 컨트롤러에서도 책임져야 할 흐름 이외의 많은 것들을 나열하고 있는 것 처럼 보이네요 :) 적절한 응집으로 필요한 정보만 노출하고 처리될 수 있도록 수정해보는 것이 어떨까요?

kingbbode reviewed (changes requested):
안녕하세요 :)
아쉬운 부분이 있어 코멘트 작성하였습니다.
확인 부탁드려요!

yelimkim98 committed:
refactor: LottoCount 객체 도입

LottoCount 객체를 도입하면서
LottosFactory 가 하는 일을 줄이고,
LottosFactory 에서 제공하던 기능을
Lottos 로 옮김.

kingbbode reviewed (approved):
리뷰가 많이 늦어져서 죄송합니다.
이만 머지하겠습니다!
수고하셨습니다 :)

kingbbode commented:
개인적인 사정으로 리뷰가 많이 늦어졌습니다. 이번에 다 못한 리뷰는 다음에 다른 방법으로 최대한 도움이 되실 수 있도록 하겠습니다! 리뷰가 늦어 죄송합니다.
